std/stdio.d

```diff
@@ -4097,17 +4143,17 @@ private struct ReadlnAppender
     }
     void putonly(char[] b)
     {
+        import std.algorithm: max;
         assert(pos == 0);   // assume this is the only put call
-        if (b.length > cap)
+        if (b.length > max(buf.length, buf.capacity))
```

Comment:
I think you should take advantage of the case `b.length <= buf.length` to avoid getting the capacity at all. This probably boils down to just calling putbuf unconditionally.